Church Role Overhaul + Grandmaster (Templar)

AllCorruptingWyrm opened the pull request:
## About The Pull Request

The following PR answers a lot of the begging we had from Church mains about why it is awful to play Acolyte and Priest. The PR introduces a LOT of QoL stuff like satchels on the Inquisitors back, some coin for Acolytes and generally god specific skills like swimming for Abyssador and so on and so forth. This was done because Templars were unruly. Along with that, this adds a Grand Templar that directly serves the Priest/ess and again, reigns in the Templars.

UPDATE: XYLIX CURSE has been untouched, for now. I'll make it do something soon but I wanna get this out the door.

This PR also touches up Inquisitor and gives them tracking, riding, and allows them to really hunt down the creechures of the nite too!

## FEEDBACK:
Please provide it, I want people to PLAY Acolyte and have actual use hence the PR, and someone to reign in unruly Templars and generally act like the last stand against the bulwark of the Church. THERE WILL be bugs.

## Why It's Good For The Game

Due to numerous requests, I've relented. A lot of this is answered in the About The Pull Request section. This has not been tested and is under a draft for that reason.

EtheoBoxxman commented on code/modules/antagonists/roguetown/villain/vampirelord.dm:
"Grand Templar," has a mispositioned comma, as well, easy fix.

EtheoBoxxman commented on code/modules/jobs/job_types/roguetown/church/puritan.dm:
I can count on one hand the amount of people who have played inquisitor in the last month. Raising the PQ will likely kill the role.

AllCorruptingWyrm replied:
Makes sense. I'll change it.

EtheoBoxxman commented on code/modules/jobs/job_types/roguetown/church/templargrand.dm:
The stats are insanely high. Even Veteran Mercenary doesnt get that much.
If this is going to be a thing, I think it should be in line with the stat array that veteran gets.

AllCorruptingWyrm replied:
This actually brings them about more or less in line with Royal Guard Captain as they get a Fortune buff as well.

@Moribun 
We can lower that down a bit.

AllCorruptingWyrm commented:
also nearly forgot the PQ changes; Yes this was done to compensate given the strength of the roles in question. Remember, we don't want shitters getting into these roles and causing issues so thats why.

AllCorruptingWyrm commented on code/modules/jobs/job_types/roguetown/church/templargrand.dm:
```diff
@@ -0,0 +1,107 @@
+		H.change_stat("intelligence", 3)
+		H.change_stat("constitution", 3)
+		H.change_stat("endurance", 3)
+		H.change_stat("fortune", 3)
```
> not even miners get fortune 3, this is bumping crit up to an extremely high chance. this is pushing deathknight stats as is. and thats an admin spawn role.

No idea why Royal Guard Captain gets Fortune then which is what these stats were based off of and what this role is to emulate.

Decoy1240 replied:
This was better then captain to be clear

AllCorruptingWyrm commented on code/modules/jobs/job_types/roguetown/church/priest.dm:
```diff
@@ -18,7 +18,7 @@
 	display_order = JDO_PRIEST
 	give_bank_account = 115
-	min_pq = 2
+	min_pq = 10
```
Sure.

AllCorruptingWyrm commented on code/modules/jobs/job_types/roguetown/church/monk.dm:
```diff
@@ -25,9 +25,10 @@
+	belt = /obj/item/storage/belt/rogue/leather/black
+	beltr = /obj/item/storage/belt/rogue/pouch/coins/rich
+	backl = /obj/item/storage/backpack/rogue/satchel
+	backpack_contents = list(/obj/item/roguekey/church = 1, /obj/item/needle/pestra = 1,) // Doktor, HEAL me!
```
yeah I can change them.

EtheoBoxxman commented:
I love the church roles and all, but there's some things about this that I just dont think would be healthy for the role and its outlook in the long term.
Grand Templar, in my opinion, is a bad idea specifically because it creates a presumed leadership role that contests the priest role. Despite the description saying they've pledged their life, there will always be the potential for dissent, and being stronger than most, if not all, combat capable roles on top of being the 'leader' of Templars would make this role antag tier for its disruptive potential.

AllCorruptingWyrm commented on code/modules/jobs/job_types/roguetown/church/monk.dm:
```diff
@@ -89,17 +90,46 @@
 			shirt =	/obj/item/clothing/suit/roguetown/shirt/undershirt/ravoxrobe			
 	if(H.mind)
-		H.mind.adjust_skillrank(/datum/skill/combat/wrestling, 5, TRUE)
+		H.mind.adjust_skillrank(/datum/skill/combat/wrestling, 3, TRUE)
```
Wrestling is a bit of a meme; Monks should not be a frontline role and that's why this has been adjusted a bit. Again, I don't mind it, but the focus of Acolyte is that they are a support arm which is why you'd wanna play them. You are trading wrestling for int and endurance for farming and crafts, giving them insane wrestling means that they are close to becoming an every man demostrated by people going warrior/rav and getting crazy good stats and balancing it out with swampweed.

Decoy1240 replied:
> Wrestling is a bit of a meme; Monks should not be a frontline role and that's why this has been adjusted a bit. Again, I don't mind it, but the focus of Acolyte is that they are a support arm which is why you'd wanna play them. You are trading wrestling for int and endurance for farming and crafts, giving them insane wrestling means that they are close to becoming an every man demostrated by people going warrior/rav and getting crazy good stats and balancing it out with swampweed.

I mean I hear you but it does have function against deadites so they should keep it just for the sake of restraining patients

Jul-Jul replied:
I believe its good for the fact incase they need to defend themselves. I see the fist fighting and wrestling skills that they prefer to non-lethally...or at least less then lethal deal with people.

AllCorruptingWyrm commented:
> I love the church roles and all, but there's some things about this that I just dont think would be healthy for the faction and its outlook in the long term. Grand Templar, in my opinion, is a bad idea specifically because it creates a presumed leadership role that contests the priest role. Despite the description saying they've pledged their life, there will always be the potential for dissent, and being stronger than most, if not all, combat capable roles on top of being the 'leader' of Templars would make this role antag tier for its disruptive potential.

This is why Priest gets their "Big Gun" which takes the form of their Xylix curse which should obliterate people when cursed by it. I will change the intro text that they are entirely beholden to the Priest and not the Church but someone needs to exist, with martial power, to keep unruly Templars in line, giving the Priest a scary lacky and ability to absolutely destroy a bad faithful/templar is necessary into keeping order. To give a comparison, on the dreaded codebase (F13) you would see the Sheriff got a gun with low TTK and AP which made people not wanna mess with the town.

Along with that, the presence of a Priest and them dying has MUCH more impact on the round by means of a zizo curse where goblins appear and kill everyone.  If the issue is with mirroring RG Captain stats then nerf RG Captain.

AllCorruptingWyrm commented:
I've made the changes as requested. Obliterated fortune but we should probably nerf royal guard captain by that point.

EtheoBoxxman commented:
> This is why Priest gets their "Big Gun" which takes the form of their Xylix curse which should obliterate people when cursed by it. 


From what I see in the code, the curse has no mechanical effects beyond publicly shaming and adding a vestigial stress label. Am I missing something?

AllCorruptingWyrm commented:
> > This is why Priest gets their "Big Gun" which takes the form of their Xylix curse which should obliterate people when cursed by it.
> 
> From what I see in the code, the curse has no mechanical effects beyond publicly shaming and adding a vestigial stress label. Am I missing something?

Thats, alarming. I'll add that to the PR then; The goal was to give them a massive fortune debuff and to make them feel weak so combat is difficult for them. Sorry, thought I added this to them. I'll add it in a bit.

Decoy1240 commented:
> I've made the changes as requested. Obliterated fortune but we should probably nerf royal guard captain by that point.

Copy the stats directly
("strength", 3)
("perception", 2)
("intelligence", 3)
("constitution", 2)
("endurance", 2)
("fortune", 2)
Captain is strong and I am fine with Templar captain having equal stats but by no means should they be better as you made them

AllCorruptingWyrm commented:
right right

I've fully mirrored the stats beyond fortune -- GT does not get a fortune buff so no more funny crits.

AllCorruptingWyrm commented:
Pretty sure I fixed the merge conflicts too.  I dread testing this.

Decoy1240 commented:
> right right
> 
> 
> 
> I've fully mirrored the stats beyond fortune -- GT does not get a fortune buff so no more funny crits.

I honestly think it shouldn't exist as to not inflate leadership roles but honestly you can keep 2 fortune I copied the stats just you should copy exactly or not at all imo. The Templars are filling the same role as guards but for the church. You can argue the grand Templar should be worse because they get devotion and guards don't which is something to consider

Decoy1240 commented:
I do want to say that as this is I don't think it will get passed this is such a massive overhaul more people then us who pay attention to code should have input. I do love the ambition and hope your vision is filled but you likely will have to make some concessions.

AllCorruptingWyrm commented:
> I do want to say that as this is I don't think it will get passed this is such a massive overhaul more people then us who pay attention to code should have input. I do love the ambition and hope your vision is filled but you likely will have to make some concessions.

Nothing has been nerfed/absolutely taken away; The roles that have had their stuff taken away got something given back in turn. Acolyte has more diversity, inquisitor and co get tracking, etc etc. Acolyte /had/ nothing till this PR, I rectified that.

Decoy1240 commented:
> > I do want to say that as this is I don't think it will get passed this is such a massive overhaul more people then us who pay attention to code should have input. I do love the ambition and hope your vision is filled but you likely will have to make some concessions.
> 
> 
> 
> Nothing has been nerfed/absolutely taken away; The roles that have had their stuff taken away got something given back in turn. Acolyte has more diversity, inquisitor and co get tracking, etc etc. Acolyte /had/ nothing till this PR, I rectified that.

I think the new role is ambitious. Priest needs more power not a captain to share power because the same issues arise if Templars don't like the priest for whatever reason

AllCorruptingWyrm commented:
> > > I do want to say that as this is I don't think it will get passed this is such a massive overhaul more people then us who pay attention to code should have input. I do love the ambition and hope your vision is filled but you likely will have to make some concessions.
> > 
> > 
> > Nothing has been nerfed/absolutely taken away; The roles that have had their stuff taken away got something given back in turn. Acolyte has more diversity, inquisitor and co get tracking, etc etc. Acolyte /had/ nothing till this PR, I rectified that.
> 
> I think the new role is ambitious. Priest needs more power not a captain to share power because the same issues arise if Templars don't like the priest for whatever reason

Issue with that is that you WILL have priests multi role and frontline and with the Zizo curse it really fucks up the round. Along with the fact that coronation and cursing is no longer a thing so people will just walk over the Church.

Troopermoores commented:
As a templar main who asked for this i appreciate the effort given, i believe the acolyte buffs are solid but an issue arises where people dont like acolytes gameplay. Even if i find it grating that you should be picking acolyte to be the holy noncombatant alot of people have a hard time to make the round entertaining for themselves. I hope that with their int gain theyll do more roleplay things like soup kitchens, wandering sermons, preaching to tribals, ect ect.

When it comes to the Grand Templar, i believe they should either restrict to middle age (if it isnt already) or reduce con and endurance down to 2 then add 1 int 1 perception. that way their clearly a successor of a templar and it reflects they are still learned through stats. while also hindering their combat ability alittle

AllCorruptingWyrm commented:
> As a templar main who asked for this i appreciate the effort given, i believe the acolyte buffs are solid but an issue arises where people dont like acolytes gameplay. Even if i find it grating that you should be picking acolyte to be the holy noncombatant alot of people have a hard time to make the round entertaining for themselves. I hope that with their int gain theyll do more roleplay things like soup kitchens, wandering sermons, preaching to tribals, ect ect.
> 
> When it comes to the Grand Templar, i believe they should either restrict to middle age (if it isnt already) or reduce con and endurance down to 2 then add 1 int 1 perception. that way their clearly a successor of a templar and it reflects they are still learned through stats. while also hindering their combat ability alittle

Done. Priest can be all patreons now, and has their big heals and revival **for now** until the other gods get useful stuff.
I fixed the spelling mistake in the Xylix curse thing and added in an extra bit that this man/woman who got cursed is to be shunned and treated like garbage essentially.

AllCorruptingWyrm commented:
Requesting additional feedback before I start testing it properly. I've heard no one complaining about it and it seems a lot of people want this merged so I'm just going to wait on maints, see what they say, and make it mergeable.

EtheoBoxxman commented on code/datums/status_effects/rogue/debuff.dm:
-5 to fortune, speed and strength for 30 minutes still feels unusually debilitating, but its much better than the -10 for an hour before.
I would personally bump it down to 10 or 15 minutes, and a -3 to speed and strength, maintaining the -5 to fortune
Keep in mind that the 30 minutes can actually be much longer than 30 minutes due to server time dilation, depends on how much the server's chugging in a given round.

The strength alone is huge, but with the speed debuff, the target also cannot defend themselves as effectively, with speed governing dodge and parry chance as well.

AllCorruptingWyrm replied:
good point. I'll change it.

EtheoBoxxman commented on code/modules/antagonists/roguetown/villain/vampirelord.dm:
```diff
@@ -909,7 +909,7 @@ GLOBAL_LIST_EMPTY(vampire_objects)
 /datum/objective/vampirelord/infiltrate/one/check_completion()
 	var/datum/game_mode/chaosmode/C = SSticker.mode
-	var/list/churchjobs = list("Priest", "Priestess", "Cleric", "Acolyte", "Templar", "Churchling", "Crusader", "Inquisitor")
+	var/list/churchjobs = list("Priest", "Priestess", "Cleric", "Acolyte", "Templar", "Churchling", "Crusader", "Grand Templar," "Inquisitor")
```
Still need the grand templar bit fixed here, the list is broken. Just change it to "Grand Templar",

EtheoBoxxman reviewed (changes requested):
Everything else looks desirable and reasonable, beyond the other comments.

AllCorruptingWyrm commented:
Going to work on these and enter bug testing phase then. Thanks for the review!

josephpatrickgreen commented on code/datums/stress/negative_events.dm:
removing someone's miracles for an hour is brutal.

AllCorruptingWyrm replied:
Nerfed/Fixed it. Its not reoving miracles, its just a debuff.

AllCorruptingWyrm commented:
OKAY, there should be no errors. I undid the changes to the Xylix curse and it compiles on my end.
Evidence that it compiles and works/shows up in game:
![image](https://github.com/user-attachments/assets/e5aef2cb-7290-4306-b811-1c0295160398) (skills)
![image](https://github.com/user-attachments/assets/5e9e7bc3-e697-4e08-8248-78d6a415993b) (role)

Grandmaster does not have a spawn point mapped in and I'll do that after some mapping PRs are done so until then, people will need to late join but with that said? Yeah we ARE GOOD to merge.
